add how-to page explaining usage of Compose provider services (#22586)
docker/docs: content/manuals/compose/how-tos/provider-services.md

## Prerequisites

- Docker Compose v2.36 or later

## What are provider services?

Provider services are a special type of service in Compose that represents platform capabilities rather than containers.
They allow you to declare dependencies on specific platform features that your application needs.

When you define a provider service in your Compose file, Compose works with the platform to provision and configure
the requested capability, making it available to your application services.

## Using provider services

To use a provider service in your Compose file, you need to:

1. Define a service with the `provider` attribute
2. Specify the `type` of provider you want to use
3. Configure any provider-specific options
4. Declare dependencies from your application services to the provider service

Here's a basic example:

```yaml
services:
  database:
    provider:
      type: awesomecloud
      options:
        type: mysql
        foo: bar  
  app:
    image: myapp 
    depends_on:
       - database
```

Notice the dedicated `provider` attribute in the `database` service.

> This attribute specifies that the service is a provider and lets you define options specific to that provider type.

Reviewer: ```suggestion
This attribute specifies that the service is managed by a provider and lets you define options specific to that provider type.
```

The `depends_on` attribute in the `app` service specifies that it depends on the `database` service.
This means that the `database` service will be started before the `app` service, allowing the provider information
to be injected into the `app` service.

## How it works

> During the `docker compose up` process, Compose identifies provider services and works with the platform to provision

Reviewer: ```suggestion
During the `docker compose up` command execution, Compose identifies services relying on providers and works with them to provision
```

> the requested capabilities. The platform then provides Compose with information about how to access the provisioned resource.

Reviewer: ```suggestion
the requested capabilities. The provider then populates Compose model with informations about how to access the provisioned resource.
```
